pkg/destroy/ibmcloud/disk.go

```diff
@@ -0,0 +1,101 @@
+package ibmcloud
+
+import (
+	"net/http"
+	"strings"
+
+	"github.com/pkg/errors"
+)
+
+func (o *ClusterUninstaller) listDisks() ([]cloudResource, error) {
+	o.Logger.Debugf("Listing volumes")
+	ctx, cancel := o.contextWithTimeout()
+	defer cancel()
+
+	options := o.vpcSvc.NewListVolumesOptions()
+	resources, _, err := o.vpcSvc.ListVolumesWithContext(ctx, options)
+	if err != nil {
+		return nil, err
+	}
+
+	result := []cloudResource{}
+	clusterIDtag := o.clusterIDFilter()
+	clusterOwnedTag := o.clusterLabelFilter()
+
+	for _, volume := range resources.Volumes {
+		useTags := strings.Join(volume.UserTags, ",")
+		if strings.Contains(useTags, clusterIDtag) || strings.Contains(useTags, clusterOwnedTag) {
+			result = append(result, cloudResource{
+				key:      *volume.ID,
+				name:     *volume.Name,
+				status:   *volume.Status,
+				typeName: "volume",
+				id:       *volume.ID,
+			})
+		}
+	}
+
+	return result, nil
+}
+
+func (o *ClusterUninstaller) deleteDisk(item cloudResource) error {
+	o.Logger.Debugf("Deleting disk %s ", item.name)
+	ctx, cancel := o.contextWithTimeout()
+	defer cancel()
+	options := o.vpcSvc.NewDeleteVolumeOptions(item.id)
+	details, err := o.vpcSvc.DeleteVolumeWithContext(ctx, options)
+
+	if err != nil {
+		return errors.Wrapf(err, "Failed to delete volume %q", item.name)
+	}
+
+	var skip = false
+
+	if err == nil && details != nil && details.StatusCode == http.StatusNoContent {
```

Comment:
It is already returning at line 48, if err != nil, I think, it's ok to remove err==nil condition from line 54.

Reply:
ok I am hold on this as we might need to change the logic after todays discussion.

Reply:
Done